test/e2e/operator_test.go

```diff
@@ -0,0 +1,290 @@
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	awsSession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/route53"
+)
+
+var kclient client.Client
+var infraConfig configv1.Infrastructure
+var dnsClient struct {
+	aws *route53.Route53
+	//TODO: add other provider clients here
+}
+var domainName string
+
+var scheme *runtime.Scheme
+
+var dialer *net.Dialer
+var httpClient *http.Client
+
+func init() {
+	scheme = kscheme.Scheme
+	if err := configv1.Install(scheme); err != nil {
+		panic(err)
+	}
+	if err := operatorv1alpha1.AddToScheme(scheme); err != nil {
+		panic(err)
+	}
+}
+
+var defaultParentDomainName string
+var defaultParentZoneID string
+
+var (
+	defaultNamespace = corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "publish-external-dns",
+		},
+	}
+	defaultBackend = corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "hello-openshift",
+			Namespace: "publish-external-dns",
+			Labels: map[string]string{
+				"name": "hello-openshift",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "hello-openshift",
+					Image: "openshift/hello-openshift",
+					Ports: []corev1.ContainerPort{
+						{
+							ContainerPort: 8080,
+							Protocol:      corev1.ProtocolTCP,
+						},
+					},
+				},
+			},
+		},
+	}
+	defaultService = corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "hello-openshift-service",
+			Namespace: "publish-external-dns",
+			Annotations: map[string]string{
+				"external-dns.mydomain.org/publish": "yes",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"name": "hello-openshift",
+			},
+			Type: corev1.ServiceTypeLoadBalancer,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: corev1.ProtocolTCP,
+					Port:     80,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8080,
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestMain(m *testing.M) {
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		fmt.Printf("failed to get kube config: %s\n", err)
+		os.Exit(1)
+	}
+
+	kubeClient, err := client.New(kubeConfig, client.Options{})
+	if err != nil {
+		fmt.Printf("failed to create kube client: %s\n", err)
+		os.Exit(1)
+	}
+	kclient = kubeClient
+
+	if err := kclient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, &infraConfig); err != nil {
+		fmt.Printf("failed to get infrastructure config: %v\n", err)
+		os.Exit(1)
+	}
+
+	defaultParentDomainName = os.Getenv("EXTDNS_PARENT_DOMAIN")
+	if defaultParentDomainName == "" {
+		fmt.Printf("No parent domain specified. Please set the environment variable EXTDNS_PARENT_DOMAIN\n")
+		os.Exit(1)
+	}
+	defaultParentZoneID = os.Getenv("EXTDNS_PARENT_ZONEID")
+	if defaultParentDomainName == "" {
```

Comment:
```suggestion
	if defaultParentZoneID == "" {
```

Reply:
good catch; I've fixed it in the latest fixup